sklearn/metrics/_plot/confusion_matrix.py

```diff
@@ -160,7 +166,248 @@ def plot(self, *, include_values=True, cmap='viridis',
         self.ax_ = ax
         return self
 
+    _common_docstring_from_methods = """labels : array-like of shape \
+                (n_classes,), default=None
+            List of labels to index the matrix. This may be used to reorder or
+            select a subset of labels. If `None` is given, those that appear at
+            least once in `y_true` or `y_pred` are used in sorted order.
 
+        sample_weight : array-like of shape (n_samples,), default=None
+            Sample weights.
+
+        normalize : {'true', 'pred', 'all'}, default=None
+            Normalizes confusion matrix over the true (rows), predicted
+            (columns) conditions or all the population. If None, confusion
+            matrix will not be normalized.
+
+        display_labels : array-like of shape (n_classes,), default=None
+            Target names used for plotting. By default, `labels` will be used
+            if it is defined, otherwise the unique labels of `y_true` and
+            `y_pred` will be used.
+
+        include_values : bool, default=True
+            Includes values in confusion matrix.
+
+        xticks_rotation : {'vertical', 'horizontal'} or float, \
+            default='horizontal'
+            Rotation of xtick labels.
+
+        values_format : str, default=None
+            Format specification for values in confusion matrix. If `None`, the
+            format specification is 'd' or '.2g' whichever is shorter.
+
+        cmap : str or matplotlib Colormap, default='viridis'
+            Colormap recognized by matplotlib.
+
+        ax : matplotlib Axes, default=None
+            Axes object to plot on. If `None`, a new figure and axes is
+            created.
+
+        colorbar : bool, default=True
+            Whether or not to add a colorbar to the plot.
+
+        Returns
+        -------
+        display : :class:`~sklearn.metrics.ConfusionMatrixDisplay`
+    """.rstrip()
+
+    @classmethod
+    @inject_docstring(common_docstring=_common_docstring_from_methods)
+    def from_estimator(
+        cls,
+        estimator,
+        X,
+        y,
+        *,
+        labels=None,
+        sample_weight=None,
+        normalize=None,
+        display_labels=None,
+        include_values=True,
+        xticks_rotation="horizontal",
+        values_format=None,
+        cmap="viridis",
+        ax=None,
+        colorbar=True,
+    ):
+        """Plot Confusion Matrix given an estimator and some data.
+
+        Read more in the :ref:`User Guide <confusion_matrix>`.
+
+        .. versionadded:: 0.24
+
+        Parameters
+        ----------
+        estimator : estimator instance
+            Fitted classifier or a fitted :class:`~sklearn.pipeline.Pipeline`
+            in which the last estimator is a classifier.
+
+        X : {{array-like, sparse matrix}} of shape (n_samples, n_features)
```

Comment:
Uhm actually it was from where the code in imbalanced-learn was contributed from :)

So double braces are the usual way for substitution using `format()` while you need to use `% ()` operator in the other solution. I don't know what's best, to be honest.

@NicolasHug Do you have any preference? If we want to use `%()`, the best would be to copy paste the code from matplotlib (minus the deprecation) and refer to it.

Reply:
I guess I'm fine with double brackets because there's no risk of wrong substitution: if we were to write `{array-like, sparse matrix}`, we'd get a KeyError at execution time.